tsl/src/compression/api.c

```diff
@@ -777,13 +778,26 @@ set_access_method(Oid relid, const char *amname)
 	};
 	bool to_hypercore = strcmp(amname, TS_HYPERCORE_TAM_NAME) == 0;
 	Oid amoid = ts_get_rel_am(relid);
+	Oid new_amoid = get_am_oid(amname, false);
 
 	/* Setting the same access method is a no-op */
-	if (amoid == get_am_oid(amname, false))
+	if (amoid == new_amoid)
 		return relid;
 
 	hypercore_alter_access_method_begin(relid, !to_hypercore);
 	AlterTableInternal(relid, list_make1(&cmd), false);
+
+#if (PG_VERSION_NUM < 150004)
+	/* Fix for PostgreSQL bug where pg_depend was not updated to reflect the
+	 * new dependency between AM and relation. See related PG fix here:
+	 * https://github.com/postgres/postgres/commit/97d89101045fac8cb36f4ef6c08526ea0841a596 */
+	if (changeDependencyFor(RelationRelationId, relid, AccessMethodRelationId, amoid, new_amoid) !=
+		1)
+		elog(ERROR,
+			 "could not change access method dependency for relation \"%s.%s\"",
+			 get_namespace_name(get_rel_namespace(relid)),
+			 get_rel_name(relid));
+#endif
```

Comment:
Since this was reported in 16beta1 it seems like it should be fixed in all GA versions of PostgreSQL after 16.0, so double-checking that it is indeed the case and this is the result. You might want to add a comment to why this is not necessary to check for later versions, but IMHO this is entirely optional since it was fixed before the GA release of the other versions.

```bash
mats@fury:~/repos/postgres$ git log --grep "Fix pg_depend entry" --oneline
97d8910104 Fix pg_depend entry to AMs after ALTER TABLE .. SET ACCESS METHOD
mats@fury:~/repos/postgres$ git tag --contains 97d8910104
REL_16_0
REL_16_1
REL_16_2
REL_16_3
REL_16_4
REL_16_5
REL_16_6
REL_16_BETA3
REL_16_RC1
REL_17_0
REL_17_1
REL_17_2
REL_17_BETA1
REL_17_BETA2
REL_17_BETA3
REL_17_RC1
```

Reply:
Thank you for double-checking.